dipy/reconst/ivim.py

```diff
@@ -0,0 +1,595 @@
+            D, neg_log_S0 = np.polyfit(bvals_split,
+                                       -np.log(data[self.gtab.bvals <=
+                                                    split_b]), 1)
+        else:
+            bvals_split = self.gtab.bvals[self.gtab.bvals >= split_b]
+            D, neg_log_S0 = np.polyfit(bvals_split,
+                                       -np.log(data[self.gtab.bvals >=
+                                                    split_b]), 1)
+
+        S0 = np.exp(-neg_log_S0)
+        return S0, D
+
+    def estimate_f_D_star(self, params_f_D_star, data, S0, D):
+        """Estimate f and D_star using the values of all the other parameters
+        obtained from a linear fit.
+
+        Parameters
+        ----------
+        params_f_D_star: array
+            An array containing the value of f and D_star.
+
+        data : array
+            Array containing the actual signal values.
+
+        S0 : float
+            The parameters S0 obtained from a linear fit.
+
+        D : float
+            The parameters D obtained from a linear fit.
+
+        Returns
+        -------
+        f : float
+           Perfusion fraction estimated from the fit.
+        D_star :
+            The value of D_star estimated from the fit.
+        """
+        gtol = self.options["gtol"]
+        ftol = self.options["ftol"]
+        xtol = self.tol
+        epsfcn = self.options["eps"]
+        maxfev = self.options["maxiter"]
+
+        if SCIPY_LESS_0_17:
+            try:
+                res = leastsq(f_D_star_error,
+                              params_f_D_star,
+                              args=(self.gtab, data, S0, D),
+                              gtol=gtol,
+                              xtol=xtol,
+                              ftol=ftol,
+                              epsfcn=epsfcn,
+                              maxfev=maxfev)
+                f, D_star = res[0]
+                return f, D_star
+            except:
+                warningMsg = "x0 obtained from linear fitting is not feasibile as "
+                warningMsg += "initial guess for leastsq. Parameters are returned only "
+                warningMsg += "from the linear fit."
+                warnings.warn(warningMsg, UserWarning)
+                f, D_star = params_f_D
+                return f, D_star
+        else:
+            try:
+                res = least_squares(f_D_star_error,
+                                    params_f_D_star,
+                                    bounds=((0., 0.), (self.bounds[1][1],
+                                                       self.bounds[1][2])),
+                                    args=(self.gtab, data, S0, D),
+                                    ftol=ftol,
+                                    xtol=xtol,
+                                    gtol=gtol,
+                                    max_nfev=maxfev)
+                f, D_star = res.x
+                return f, D_star
+            except:
```

Comment:
Same here -- what exception class would be acceptable to run the next block of code?